### PR TITLE
Fix script to create and import TestRail cases from GitHub issues

### LINKED ISSUE #2: RDD generation fails when a theme issue is not a zenhub issue
### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

RDD fails when a theme issue is not a zenhub issue.

### 🕵️ Expected behavior

We expect the generation to finish without failure.

### 📜 To Reproduce

See action https://github.com/NASA-PDS/nasa-pds.github.io/actions/runs/6329387432/job/17189559877

It happens for ticket https://github.com/NASA-PDS/software-issues-repo/issues/78

### 🖥 Environment Info

As used in github actions.

### 📚 Version of Software Used

- Version of this software 1.0.0

### 🩺 Test Data / Additional context

_No response_

### 🦄 Related requirements

_No response_

### ⚙️ Engineering Details

_No response_

### PR DESCRIPTION
**Pull request by c-suh**
## 🗒️ Summary

Relevant changes for the specific problem are in the commit titled "[Add helper method to parse responses for TestRail v8 and v10+](https://github.com/NASA-PDS/lasso-issues/commit/d55f266c65611e4efee5b8bb7f6f2f7ca2f0c6ca)".

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: ___ %

## ⚙️ Test Data and/or Report

To manually test:
1. Set up lasso issues to run on your local machine and communicate with TestRail (e.g., install lasso-issues, have your TestRail token on hand)
2. Run the command (e.g., `pds-issues --format csv --build B18 --testrail-url <testrail_url> --testrail-user-email <testrail_user_email> --testrail-user-token <testrail_user_token>`)
3. Watch the output

## ♻️ Related Issues

Resolves https://github.com/NASA-PDS/lasso-issues/issues/59

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: .github/workflows/issue-project-automation.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      # Multiple projects: project_numbers: "6,22"
+      project_numbers: "6"
+    secrets:
+      # IMPORTANT: GITHUB_TOKEN does not have permission for org projects
+      # You must create a PAT with 'project' scope and add it as ORG_PROJECT_PAT
+      # See workflow-templates/README.md for setup instructions
+      gh_token: ${{ secrets.ORG_PROJECT_PAT }}
+
+  # When a label is added or removed, handle build labels and sprint-backlog automation
+  handle-label:
```
[the remark this answers is not in the document]
Comment (c-suh) [reply .github/workflows/issue-project-automation.yml:26]: Ignoring as these are files folded in from merging `main` branch, not files changed to address the issue at hand.

File: src/lasso/issues/issues/RstRddReport.py [hunk 2]
```diff
@@ -1294,6 +1294,10 @@ def __init__(
         self._filename = filename
 
         from requests.auth import HTTPBasicAuth
+        import urllib3
+
+        # Suppress InsecureRequestWarnings for TestRail SSL verification
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
         self._basic_auth = HTTPBasicAuth(testrail_user_email, testrail_user_token)
```
[the remark this answers is not in the document]
Comment (c-suh) [reply src/lasso/issues/issues/RstRddReport.py:1303]: Adequately addressed with commit https://github.com/NASA-PDS/lasso-issues/pull/60/commits/afbaae56ecd90d2583cfe5388e63a47fd9be6f2c.

File: src/lasso/issues/issues/RstRddReport.py [hunk 3]
```diff
@@ -1302,30 +1306,65 @@ def __init__(
...
 
     def create(self, repos):
         """Create."""
+        print("=" * 80)
+        print("SECURITY NOTICE:")
+        print("This script disables SSL certificate verification (verify=False) for all")
+        print("TestRail API requests and suppresses InsecureRequestWarning messages.")
+        print("This configuration should only be used in trusted network environments.")
+        print("=" * 80)
+        print()
```
Comment (nutjob4life) [inline src/lasso/issues/issues/RstRddReport.py]: This is a great warning, however diagnostic messages (including 🚨ALL🚔CAPS👮SECURITY🚓NOTICES😆) should go to `file=sys.stderr`.

File: src/lasso/issues/issues/RstRddReport.py [hunk 4]
```diff
@@ -1302,30 +1306,65 @@ def __init__(
...
             verify=False,
         )
-        # depending on testrail version
-        section_response_json = section_response.json()
-        # if true for testrail version 8
-        section_list = (
-            section_response_json["sections"] if "sections" in section_response.json() else section_response_json
-        )
+
+        if section_response.status_code != 200:
```
Comment (nutjob4life) [inline src/lasso/issues/issues/RstRddReport.py]: You can replace `200` with `http.HTTPStatus.OK` or even handle all the "ok" codes at once with this Python-Requests–specific conditional:
```python
if not section_response.ok:
    raise RuntimeError(…)
```

**Review by nutjob4life — CHANGES_REQUESTED** [review]
This looks nice! It's a solid and practical approach for TestRail v10 compatibility. I did notice 2 minor things (see interspersed comments).

I also notice that `tox` no longer completes in the repository:
```console
$ cd /tmp
$ git clone --quiet --branch testrail https://github.com/NASA-PDS/lasso-issues.git
$ cd lasso-issues
$ python3.13 -m venv .venv
$ . .venv/bin/activate
(.venv) bash-5.3$ pip install --quiet --editable '.[dev]'
…
$ pre-commit install; pre-commit install -t pre-push; pre-commit install -t prepare-commit-msg; pre-commit install -t commit-msg
pre-commit installed at .git/hooks/pre-commit
pre-commit installed at .git/hooks/pre-push
pre-commit installed at .git/hooks/prepare-commit-msg
pre-commit installed at .git/hooks/commit-msg
$ tox
…
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing examples/b18_schedule_prep_spreadsheet.csv

Check that executables have shebangs.................(no files to check)Skipped
Check for merge conflicts................................................Passed
Debug Statements (Python)................................................Passed
Check Yaml...............................................................Passed
Reorder python imports...................................................Passed
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

src/lasso/issues/issues/RstRddReport.py:1310:1: D212 Multi-line docstring summary should start at the first line
src/lasso/issues/issues/issues.py:29:1: E303 too many blank lines (3)
src/lasso/issues/issues/utils.py:102:42: E741 ambiguous variable name 'l'
src/lasso/issues/issues/utils.py:107:27: E741 ambiguous variable name 'l'
src/lasso/issues/issues/utils.py:111:52: E741 ambiguous variable name 'l'

Detect secrets...........................................................Passed
lint: exit 1 (2.33 seconds) /private/tmp/lasso-issues> python -m pre_commit run --color=always --all pid=54242
  py313: OK (26.36=setup[15.93]+cmd[10.43] seconds)
  docs: OK (12.84=setup[11.57]+cmd[1.27] seconds)
  lint: FAIL code 1 (3.56=setup[1.23]+cmd[2.33] seconds)
  evaluation failed :( (42.92 seconds)
```
Might be nice to get that fixed 😉

**Comment by c-suh** [comment]
@nutjob4life thank you, as always! I believe I've addressed your points and have requested you re-review.

File: src/lasso/issues/issues/RstRddReport.py [hunk 5]
```diff
@@ -1446,22 +1485,60 @@ def add_repo(self, repo):
...
-            # if true for testrail version 8
-            cases = cases_json["cases"] if "cases" in cases_json else cases_json
-            for case in cases:
-                refs = case["refs"].split(",")
-                issue_ref = [ref for ref in refs if "#" in ref][0]
-                self._logger.debug("looking at the pre-existing issue ref %s", issue_ref)
-                ac_ref = [ref for ref in refs if ref.startswith("AC")][0]
-                self._current_repo_existing_cases[f"{issue_ref}-{ac_ref}"] = case["id"]
+
+            if cases_resp.status_code != 200:
```
Comment (nutjob4life) [inline src/lasso/issues/issues/RstRddReport.py]: Just noticed here we can also use the `ok` attribute:
```python
if not cases_repo.ok:
    …
```

**Review by nutjob4life — APPROVED** [review]
Code delta: ✓
Issues addressed: ✓
One more `200` found: ✓
You want to replace it with `.ok`: up to you!
Tox: ✓
Approval: ✅
Tox details:
```
…
  py313: OK (11.37=setup[6.51]+cmd[4.87] seconds)
  docs: OK (1.03=setup[0.39]+cmd[0.64] seconds)
  lint: OK (2.51=setup[0.01]+cmd[2.50] seconds)
  congratulations :) (15.05 seconds)
```

File: src/lasso/issues/issues/RstRddReport.py (hunk 5 above)
Comment (c-suh) [reply src/lasso/issues/issues/RstRddReport.py]: that's on me; I should've searched for "200"

Comment (nutjob4life) [reply src/lasso/issues/issues/RstRddReport.py]: But we got it eventually! It's a team effort 🏆
